test/test_core.h

```diff
@@ -25,59 +25,61 @@
     SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 #pragma once
+#include "string"
+#include "vector"
 #include "gtest/gtest.h"
 #include "testhelper.h"
 
 #include "tiny_dnn/tiny_dnn.h"
 
 #if defined(USE_OPENCL) || defined(USE_CUDA)
 #include "third_party/CLCudaAPI/clpp11.h"
-#endif //defined(USE_OPENCL) || defined(USE_CUDA)
+#endif  // defined(USE_OPENCL) || defined(USE_CUDA)
 
 namespace tiny_dnn {
 
 #if defined(USE_OPENCL) || defined(USE_CUDA)
 device_t device_type(size_t &platform, size_t &device) {
-    // check which platforms are available
-    auto platforms = CLCudaAPI::GetAllPlatforms();
+  // check which platforms are available
+  auto platforms = CLCudaAPI::GetAllPlatforms();
 
-    // if no platforms - return -1
-    if (platforms.size() == 0) {
-        return device_t::NONE;
+  // if no platforms - return -1
+  if (platforms.size() == 0) {
+    return device_t::NONE;
+  }
+
+  for (auto p = platforms.begin(); p != platforms.end(); ++p)
+    for (size_t d = 0; d < p->NumDevices(); ++d) {
+      auto dev = CLCudaAPI::Device(*p, d);
+      if (dev.Type() == "GPU") {
+        platform = p - platforms.begin();
+        device = d;
+        return device_t::GPU;
+      }
     }
 
-    for (auto p = platforms.begin(); p != platforms.end(); ++p)
-        for (size_t d = 0; d < p->NumDevices(); ++d) {
-            auto dev = CLCudaAPI::Device(*p, d);
-            if (dev.Type() == "GPU") {
-                platform = p - platforms.begin();
-                device = d;
-                return device_t::GPU;
-            }
-        }
-
-    for (auto p = platforms.begin(); p != platforms.end(); ++p)
-        for (size_t d = 0; d < p->NumDevices(); ++d) {
-            auto dev = CLCudaAPI::Device(*p, d);
-            if (dev.Type() == "CPU") {
-                platform = p - platforms.begin();
-                device = d;
-                return device_t::CPU;
-            }
-        }
+  for (auto p = platforms.begin(); p != platforms.end(); ++p)
+    for (size_t d = 0; d < p->NumDevices(); ++d) {
+      auto dev = CLCudaAPI::Device(*p, d);
+      if (dev.Type() == "CPU") {
+        platform = p - platforms.begin();
+        device = d;
+        return device_t::CPU;
+      }
+    }
 
-    // no CPUs or GPUs
-    return device_t::NONE;
+  // no CPUs or GPUs
+  return device_t::NONE;
 }
 
 #define TINY_DNN_GET_DEVICE_AND_PLATFORM       \
-    cnn_size_t cl_platform = 0, cl_device = 0; \
+    size_t cl_platform = 0, cl_device = 0; \
     device_t device = device_type(cl_platform, cl_device);
```

Comment:
@Randl could you open a ticket in order to continue with the discussion. I'll merge this PR